source/camera/CullingManager.cpp

```diff
@@ -0,0 +1,253 @@
+    // create a slope -- a set of left and rigth path
+    mWalk.convex_hull();
+    mWalk.buildSlopes();
+
+    OD_LOG_DBG( mOldWalk.debug());
+    OD_LOG_DBG( mWalk.debug());        
+
+    // reset index pointers to the begging of collections
+    mOldWalk.prepareWalk();
+    mWalk.prepareWalk();
+
+    newBashAndSplashTiles(HIDE);
+    showAllTiles();
+
+}
+
+void CullingManager::hideAllTiles(void)
+{
+    GameMap* gm = mCm->getGameMap();
+    if (!gm)
+        return;
+
+    for (int jj = 0; jj < gm->getMapSizeY() ; ++jj)
+    {
+        for (int ii = 0; ii < gm->getMapSizeX(); ++ii)
+        {
+            RenderManager::getSingleton().rrDetachEntity(gm->getTile(ii, jj));
+        }
+    }
+}
+
+void CullingManager::showAllTiles(void)
+{
+    GameMap* gm = mCm->getGameMap();
+    if (!gm)
+        return;
+
+    for (int jj = 0; jj < gm->getMapSizeY() ; ++jj)
+    {
+        for (int ii = 0; ii < gm->getMapSizeX(); ++ii)
+        {
+            RenderManager::getSingleton().rrAttachEntity(gm->getTile(ii, jj));
+        }
+    }
+}
+
+void CullingManager::newBashAndSplashTiles(int64_t mode){
+    int64_t xxLeftOld = mOldWalk.getTopLeftVertex().x;
+    int64_t xxRightOld= mOldWalk.getTopRightVertex().x;
+    int64_t xxLeft = mWalk.getTopLeftVertex().x;
+    int64_t xxRight= mWalk.getTopRightVertex().x;
+    int64_t xxp, yyp;
+    std::stringstream ss;
+    int64_t bb = (( std::min(mWalk.getBottomLeftVertex().y , mOldWalk.getBottomRightVertex().y) >> PRECISION_DIGITS) - 2) << PRECISION_DIGITS;
+
+    for (int64_t yy = ((std::max(mWalk.getTopLeftVertex().y , mOldWalk.getTopRightVertex().y  ) >> PRECISION_DIGITS) + 2) << PRECISION_DIGITS;  yy >= bb; yy -= UNIT)
+    {
+        mOldWalk.notifyOnMoveDown(yy);
+        mWalk.notifyOnMoveDown(yy);
+        xxLeft = mWalk.getCurrentXLeft(yy);
+        xxLeftOld = mOldWalk.getCurrentXLeft(yy);
+        xxRight = mWalk.getCurrentXRight(yy);
+        xxRightOld = mOldWalk.getCurrentXRight(yy);
+   
+        int64_t mm = ((std::min(xxLeft, xxLeftOld) >> PRECISION_DIGITS) << PRECISION_DIGITS) ;
+        if(std::min(xxLeft, xxLeftOld) < std::max(xxRight,xxRightOld) )
+        {
+            for (int64_t xx = mm ; xx <= std::max(xxRight,xxRightOld) ; xx+= UNIT)
+            {
+                bool bash = (xx >= xxLeftOld && xx <= xxRightOld && (yy >= mOldWalk.getBottomLeftVertex().y) && yy <= mOldWalk.getTopLeftVertex().y);
+                bool splash = (xx >= xxLeft && xx <= xxRight && (yy >= mWalk.getBottomLeftVertex().y) && yy <= mWalk.getTopLeftVertex().y);
+
+                xxp = ((xx >>PRECISION_DIGITS));
+                yyp = ((yy >>PRECISION_DIGITS));
+                GameMap* gm = mCm->getGameMap();
+                if(bash && splash && (mode & HIDE) && (mode & SHOW))
+                {
+                    // Nothing
+                }
+
+                else if (gm && bash && (mode & HIDE) && xxp>=0 && yyp>= 0 && xxp<mCm->getGameMap()->getMapSizeX() && yyp < mCm->getGameMap()->getMapSizeY())
```

Comment:
At first I laughed, but after further investigation I was able to read that code  . 
If I could have something to say about my code that would be the only site not to change it because I hate using continue and break .